keylime/revocation_notifier.py

```diff
@@ -216,7 +216,7 @@ def await_notifications(callback, revocation_cert_path):
 
     while True:
         rawbody = mysock.recv()
-        body = json.loads(rawbody)
+        body = json.loads(rawbody)  # type: ignore
```

Comment:
Can we instead add a type hint to body?

Reply:
How? Like this?
``` 
              rawbody: bytes = mysock.recv()
```

I get this error then:
```
/home/stefanb/keylime/keylime/revocation_notifier.py
  /home/stefanb/keylime/keylime/revocation_notifier.py:218:26 - error: Expression of type "Frame | ByteString" cannot be assigned to declared type "bytes"
    Type "Frame | ByteString" cannot be assigned to type "bytes"
      "ByteString" is incompatible with "bytes" (reportGeneralTypeIssues)
```

Reply:
Ah ok. Can you then use `typing.cast()` like this:
```
              rawbody: bytes = cast(bytes, mysock.recv())
```

@aplanas is there otherwise a better way to eliminate the need for ` # type:ignore`?

Reply:
Yes, this works. The famous isinstance() check as makes pyright happy even though it seems to know that rawbody is ByteString or Frame.
```
    while True:
        rawbody = mysock.recv()
        assert isinstance(rawbody, bytes)
        body = json.loads(rawbody)
```

Reply:
3 choices. What's the best one (here)?

Reply:
I like the cast option most, but I couldn't figure out what the right way is according to the Python standards.

Reply:
The real problem IMO is that for some reason pyright thinks `mysock.recv()` returns `Frame | ByteString`. Where does that come from?